libs/utils/src/celix_hash_map.c

```diff
@@ -339,21 +327,27 @@ static bool celix_hashMap_remove(celix_hash_map_t* map, const char* strKey, long
         visit = visit->next;
     }
     if (removedEntry != NULL) {
+        char* removedKey = NULL;
+        if (map->keyType == CELIX_HASH_MAP_STRING_KEY) {
+            removedKey = (char*)removedEntry->key.strKey;
+        }
         celix_hashMap_destroyRemovedEntry(map, removedEntry);
+        if (removedKey) {
+            celix_hashMap_destroyRemovedKey(map, removedKey);
+        }
         return true;
     }
     return false;
 }
 
-static void celix_hashMap_init(
+celix_status_t celix_hashMap_init(
         celix_hash_map_t* map,
         celix_hash_map_key_type_e keyType,
         unsigned int initialCapacity,
         double loadFactor,
         unsigned int (*hashKeyFn)(const celix_hash_map_key_t*),
         bool (*equalsKeyFn)(const celix_hash_map_key_t*, const celix_hash_map_key_t*)) {
     map->loadFactor = loadFactor;
```

Comment:
Interesting. The load factor of 0.75 was and still is used in the deprecated hash map. I did reading up on hash maps when I created the long/string hash map, but apparently missed this and reused the 0.75 factor. 

I spent some hours tweaking the max load factor and hash functions combined with the already existing hash map benchmarks. Based on that result I changed the default long hash, refactored some code to prevent some function calls and changed the max load factor. 

Max load factor is now 5.0 and the long hash function is also changed (uses a prime number for an overall better distribution in the hash map).

I quite certain more improvements are possible for the long hash map and this can also been seen in the benchmark reference (`std::unordered_map with a long key`). 
The celix string hash map performance, for finding an entry in the hash map, is quite close to the std::unordered_map with a string key.

Reply:
Here are the load factor benchmark result (on my machine, using a release build type and cpupower on performance):

# Hash3 , max load 0.75
```
-----------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                               Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------------------
LongHashmapBenchmark_findEntryFromStdMap/10/process_time/real_time                  0.421 ns        0.421 ns   1000000000 items_per_second=2.37362G/s
LongHashmapBenchmark_findEntryFromStdMap/100/process_time/real_time                 0.421 ns        0.421 ns   1000000000 items_per_second=2.37341G/s
LongHashmapBenchmark_findEntryFromStdMap/1000/process_time/real_time                0.422 ns        0.422 ns   1000000000 items_per_second=2.36961G/s
LongHashmapBenchmark_findEntryFromStdMap/10000/process_time/real_time               0.421 ns        0.421 ns   1000000000 items_per_second=2.37266G/s
LongHashmapBenchmark_findEntryFromStdMap/100000/process_time/real_time              0.422 ns        0.422 ns   1000000000 items_per_second=2.3715G/s
LongHashmapBenchmark_findEntryFromStdMap/1000000/process_time/real_time             0.422 ns        0.422 ns   1000000000 items_per_second=2.3694G/s
LongHashmapBenchmark_findEntryFromCelixMap/10/process_time/real_time                 2.32 ns         2.32 ns    302202596 averageNrOfEntriesPerBucket=0.625 items_per_second=431.121M/s nrOfBuckets=16 resizeCount=0 stdDeviationNrOfEntriesPerBucket=0.856957
LongHashmapBenchmark_findEntryFromCelixMap/100/process_time/real_time                2.58 ns         2.58 ns    272789300 averageNrOfEntriesPerBucket=0.390625 items_per_second=387.522M/s nrOfBuckets=256 resizeCount=4 stdDeviationNrOfEntriesPerBucket=0.687322
LongHashmapBenchmark_findEntryFromCelixMap/1000/process_time/real_time               2.37 ns         2.37 ns    295845710 averageNrOfEntriesPerBucket=0.488281 items_per_second=421.411M/s nrOfBuckets=2.048k resizeCount=7 stdDeviationNrOfEntriesPerBucket=0.695872
LongHashmapBenchmark_findEntryFromCelixMap/10000/process_time/real_time              2.39 ns         2.39 ns    294741774 averageNrOfEntriesPerBucket=0.610352 items_per_second=417.892M/s nrOfBuckets=16.384k resizeCount=10 stdDeviationNrOfEntriesPerBucket=0.784076
LongHashmapBenchmark_findEntryFromCelixMap/100000/process_time/real_time             2.40 ns         2.40 ns    297425736 averageNrOfEntriesPerBucket=0.718061 items_per_second=417.292M/s nrOfBuckets=139.264k resizeCount=22 stdDeviationNrOfEntriesPerBucket=0.846435
LongHashmapBenchmark_findEntryFromCelixMap/1000000/process_time/real_time            2.27 ns         2.27 ns    308634832 averageNrOfEntriesPerBucket=0.747751 items_per_second=440.035M/s nrOfBuckets=1.33734M resizeCount=139 stdDeviationNrOfEntriesPerBucket=0.863872
LongHashmapBenchmark_findEntryFromDeprecatedMap/10/process_time/real_time            3.53 ns         3.53 ns    198927031 items_per_second=283.682M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100/process_time/real_time           3.52 ns         3.52 ns    198043432 items_per_second=283.711M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000/process_time/real_time          3.53 ns         3.53 ns    195734925 items_per_second=283.213M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/10000/process_time/real_time         3.53 ns         3.53 ns    197803895 items_per_second=283.436M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100000/process_time/real_time        3.52 ns         3.52 ns    198613048 items_per_second=284.045M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000000/process_time/real_time       3.50 ns         3.50 ns    199963559 items_per_second=285.326M/s
```

# Hash3 , max load 3
```
-----------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                               Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------------------
LongHashmapBenchmark_findEntryFromStdMap/10/process_time/real_time                  0.425 ns        0.425 ns   1000000000 items_per_second=2.35061G/s
LongHashmapBenchmark_findEntryFromStdMap/100/process_time/real_time                 0.426 ns        0.426 ns   1000000000 items_per_second=2.34699G/s
LongHashmapBenchmark_findEntryFromStdMap/1000/process_time/real_time                0.426 ns        0.426 ns   1000000000 items_per_second=2.34825G/s
LongHashmapBenchmark_findEntryFromStdMap/10000/process_time/real_time               0.426 ns        0.426 ns   1000000000 items_per_second=2.34747G/s
LongHashmapBenchmark_findEntryFromStdMap/100000/process_time/real_time              0.426 ns        0.426 ns   1000000000 items_per_second=2.3473G/s
LongHashmapBenchmark_findEntryFromStdMap/1000000/process_time/real_time             0.437 ns        0.437 ns   1000000000 items_per_second=2.28811G/s
LongHashmapBenchmark_findEntryFromCelixMap/10/process_time/real_time                 2.34 ns         2.34 ns    299224720 averageNrOfEntriesPerBucket=0.625 items_per_second=428.114M/s nrOfBuckets=16 resizeCount=0 stdDeviationNrOfEntriesPerBucket=0.856957
LongHashmapBenchmark_findEntryFromCelixMap/100/process_time/real_time                2.76 ns         2.76 ns    252886933 averageNrOfEntriesPerBucket=1.5625 items_per_second=362.06M/s nrOfBuckets=64 resizeCount=2 stdDeviationNrOfEntriesPerBucket=1.26089
LongHashmapBenchmark_findEntryFromCelixMap/1000/process_time/real_time               2.34 ns         2.34 ns    299386105 averageNrOfEntriesPerBucket=1.95312 items_per_second=428.21M/s nrOfBuckets=512 resizeCount=5 stdDeviationNrOfEntriesPerBucket=1.47033
LongHashmapBenchmark_findEntryFromCelixMap/10000/process_time/real_time              2.55 ns         2.55 ns    274628326 averageNrOfEntriesPerBucket=2.44141 items_per_second=392.449M/s nrOfBuckets=4.096k resizeCount=8 stdDeviationNrOfEntriesPerBucket=1.56031
LongHashmapBenchmark_findEntryFromCelixMap/100000/process_time/real_time             2.34 ns         2.34 ns    299777246 averageNrOfEntriesPerBucket=2.71267 items_per_second=428.166M/s nrOfBuckets=36.864k resizeCount=12 stdDeviationNrOfEntriesPerBucket=1.64409
LongHashmapBenchmark_findEntryFromCelixMap/1000000/process_time/real_time            2.33 ns         2.33 ns    299923251 averageNrOfEntriesPerBucket=2.99559 items_per_second=428.295M/s nrOfBuckets=333.824k resizeCount=41 stdDeviationNrOfEntriesPerBucket=1.73026
LongHashmapBenchmark_findEntryFromDeprecatedMap/10/process_time/real_time            3.63 ns         3.63 ns    192852998 items_per_second=275.666M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100/process_time/real_time           3.63 ns         3.63 ns    192824656 items_per_second=275.619M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000/process_time/real_time          3.63 ns         3.63 ns    192825416 items_per_second=275.698M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/10000/process_time/real_time         3.63 ns         3.63 ns    192885922 items_per_second=275.674M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100000/process_time/real_time        3.63 ns         3.63 ns    193014477 items_per_second=275.727M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000000/process_time/real_time       3.66 ns         3.66 ns    191065573 items_per_second=272.882M/s
```

# Hash3, max load 5
```
-----------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                               Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------------------
LongHashmapBenchmark_findEntryFromStdMap/10/process_time/real_time                  0.414 ns        0.414 ns   1000000000 items_per_second=2.4128G/s
LongHashmapBenchmark_findEntryFromStdMap/100/process_time/real_time                 0.414 ns        0.414 ns   1000000000 items_per_second=2.41719G/s
LongHashmapBenchmark_findEntryFromStdMap/1000/process_time/real_time                0.414 ns        0.414 ns   1000000000 items_per_second=2.41582G/s
LongHashmapBenchmark_findEntryFromStdMap/10000/process_time/real_time               0.414 ns        0.414 ns   1000000000 items_per_second=2.41681G/s
LongHashmapBenchmark_findEntryFromStdMap/100000/process_time/real_time              0.414 ns        0.414 ns   1000000000 items_per_second=2.41521G/s
LongHashmapBenchmark_findEntryFromStdMap/1000000/process_time/real_time             0.414 ns        0.414 ns   1000000000 items_per_second=2.41732G/s
LongHashmapBenchmark_findEntryFromCelixMap/10/process_time/real_time                 2.26 ns         2.26 ns    307909253 averageNrOfEntriesPerBucket=0.625 items_per_second=441.585M/s nrOfBuckets=16 resizeCount=0 stdDeviationNrOfEntriesPerBucket=0.856957
LongHashmapBenchmark_findEntryFromCelixMap/100/process_time/real_time                3.39 ns         3.39 ns    211113789 averageNrOfEntriesPerBucket=3.125 items_per_second=295.319M/s nrOfBuckets=32 resizeCount=1 stdDeviationNrOfEntriesPerBucket=1.86665
LongHashmapBenchmark_findEntryFromCelixMap/1000/process_time/real_time               2.48 ns         2.48 ns    281288298 averageNrOfEntriesPerBucket=3.90625 items_per_second=403.557M/s nrOfBuckets=256 resizeCount=4 stdDeviationNrOfEntriesPerBucket=2.16123
LongHashmapBenchmark_findEntryFromCelixMap/10000/process_time/real_time              3.31 ns         3.31 ns    211375732 averageNrOfEntriesPerBucket=4.88281 items_per_second=302.161M/s nrOfBuckets=2.048k resizeCount=7 stdDeviationNrOfEntriesPerBucket=2.2035
LongHashmapBenchmark_findEntryFromCelixMap/100000/process_time/real_time             2.48 ns         2.48 ns    282405576 averageNrOfEntriesPerBucket=3.75601 items_per_second=403.207M/s nrOfBuckets=26.624k resizeCount=11 stdDeviationNrOfEntriesPerBucket=1.92598
LongHashmapBenchmark_findEntryFromCelixMap/1000000/process_time/real_time            2.90 ns         2.90 ns    241719478 averageNrOfEntriesPerBucket=4.98246 items_per_second=344.381M/s nrOfBuckets=200.704k resizeCount=28 stdDeviationNrOfEntriesPerBucket=2.23011
LongHashmapBenchmark_findEntryFromDeprecatedMap/10/process_time/real_time            3.54 ns         3.54 ns    197602959 items_per_second=282.865M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100/process_time/real_time           3.54 ns         3.53 ns    197704646 items_per_second=282.885M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000/process_time/real_time          3.53 ns         3.53 ns    197684458 items_per_second=282.891M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/10000/process_time/real_time         3.54 ns         3.54 ns    197731536 items_per_second=282.88M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100000/process_time/real_time        3.54 ns         3.54 ns    197896830 items_per_second=282.817M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000000/process_time/real_time       3.54 ns         3.54 ns    198110019 items_per_second=282.856M/s
```

# Hash3, max load 10
```
-----------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                               Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------------------
LongHashmapBenchmark_findEntryFromStdMap/10/process_time/real_time                  0.435 ns        0.435 ns   1000000000 items_per_second=2.29977G/s
LongHashmapBenchmark_findEntryFromStdMap/100/process_time/real_time                 0.432 ns        0.432 ns   1000000000 items_per_second=2.31494G/s
LongHashmapBenchmark_findEntryFromStdMap/1000/process_time/real_time                0.426 ns        0.426 ns   1000000000 items_per_second=2.34589G/s
LongHashmapBenchmark_findEntryFromStdMap/10000/process_time/real_time               0.426 ns        0.426 ns   1000000000 items_per_second=2.34525G/s
LongHashmapBenchmark_findEntryFromStdMap/100000/process_time/real_time              0.426 ns        0.426 ns   1000000000 items_per_second=2.34972G/s
LongHashmapBenchmark_findEntryFromStdMap/1000000/process_time/real_time             0.425 ns        0.425 ns   1000000000 items_per_second=2.35186G/s
LongHashmapBenchmark_findEntryFromCelixMap/10/process_time/real_time                 2.33 ns         2.33 ns    299579738 averageNrOfEntriesPerBucket=0.625 items_per_second=428.363M/s nrOfBuckets=16 resizeCount=0 stdDeviationNrOfEntriesPerBucket=0.856957
LongHashmapBenchmark_findEntryFromCelixMap/100/process_time/real_time                2.57 ns         2.57 ns    274525405 averageNrOfEntriesPerBucket=6.25 items_per_second=388.373M/s nrOfBuckets=16 resizeCount=0 stdDeviationNrOfEntriesPerBucket=2.53722
LongHashmapBenchmark_findEntryFromCelixMap/1000/process_time/real_time               3.03 ns         3.03 ns    219679035 averageNrOfEntriesPerBucket=7.8125 items_per_second=329.912M/s nrOfBuckets=128 resizeCount=3 stdDeviationNrOfEntriesPerBucket=3.12687
LongHashmapBenchmark_findEntryFromCelixMap/10000/process_time/real_time              8.02 ns         8.02 ns     81140813 averageNrOfEntriesPerBucket=9.76562 items_per_second=124.642M/s nrOfBuckets=1024 resizeCount=6 stdDeviationNrOfEntriesPerBucket=3.15172
LongHashmapBenchmark_findEntryFromCelixMap/100000/process_time/real_time             2.55 ns         2.55 ns    274930765 averageNrOfEntriesPerBucket=6.10352 items_per_second=392.633M/s nrOfBuckets=16.384k resizeCount=10 stdDeviationNrOfEntriesPerBucket=2.46868
LongHashmapBenchmark_findEntryFromCelixMap/1000000/process_time/real_time            3.18 ns         3.18 ns    220016340 averageNrOfEntriesPerBucket=9.21285 items_per_second=313.99M/s nrOfBuckets=108.544k resizeCount=19 stdDeviationNrOfEntriesPerBucket=3.02688
LongHashmapBenchmark_findEntryFromDeprecatedMap/10/process_time/real_time            3.62 ns         3.62 ns    193278219 items_per_second=276.196M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100/process_time/real_time           3.62 ns         3.62 ns    193216614 items_per_second=276.198M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000/process_time/real_time          3.62 ns         3.62 ns    193269813 items_per_second=276.076M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/10000/process_time/real_time         3.62 ns         3.62 ns    193179709 items_per_second=276.217M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100000/process_time/real_time        3.62 ns         3.62 ns    193208642 items_per_second=276.177M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000000/process_time/real_time       3.62 ns         3.62 ns    193566533 items_per_second=276.364M/s

Reply:
Here are the hash function benchmark results (on my machine, using a release build type and cpupower on performance):

hash1 is the long hash function used in celix_long_hash_map at the start of this PR
hash2 is the long hash function used in the deprecated hashmap 
hash3 is a altered long hash function which uses a prime.

I also had a short look into [murmur3](https://github.com/logrhythm/MurMurHash/blob/master/src/MurMurHash.cpp) and [FNV](https://github.com/fnvhash/libfnv/tree/master) hash, but for the moment I think this requires too much work (technical and license wise)

# Hash1, max load 5
```
-----------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                               Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------------------
LongHashmapBenchmark_findEntryFromStdMap/10/process_time/real_time                  0.432 ns        0.432 ns   1000000000 items_per_second=2.31246G/s
LongHashmapBenchmark_findEntryFromStdMap/100/process_time/real_time                 0.433 ns        0.433 ns   1000000000 items_per_second=2.31041G/s
LongHashmapBenchmark_findEntryFromStdMap/1000/process_time/real_time                0.433 ns        0.433 ns   1000000000 items_per_second=2.31151G/s
LongHashmapBenchmark_findEntryFromStdMap/10000/process_time/real_time               0.434 ns        0.434 ns   1000000000 items_per_second=2.30551G/s
LongHashmapBenchmark_findEntryFromStdMap/100000/process_time/real_time              0.433 ns        0.433 ns   1000000000 items_per_second=2.31037G/s
LongHashmapBenchmark_findEntryFromStdMap/1000000/process_time/real_time             0.433 ns        0.432 ns   1000000000 items_per_second=2.30713G/s
LongHashmapBenchmark_findEntryFromCelixMap/10/process_time/real_time                 2.38 ns         2.38 ns    293482551 averageNrOfEntriesPerBucket=0.625 items_per_second=420.077M/s nrOfBuckets=16 resizeCount=0 stdDeviationNrOfEntriesPerBucket=0.856957
LongHashmapBenchmark_findEntryFromCelixMap/100/process_time/real_time                2.64 ns         2.63 ns    268351618 averageNrOfEntriesPerBucket=3.125 items_per_second=379.474M/s nrOfBuckets=32 resizeCount=1 stdDeviationNrOfEntriesPerBucket=1.91621
LongHashmapBenchmark_findEntryFromCelixMap/1000/process_time/real_time               2.16 ns         2.16 ns    323130939 averageNrOfEntriesPerBucket=3.90625 items_per_second=462.94M/s nrOfBuckets=256 resizeCount=4 stdDeviationNrOfEntriesPerBucket=1.80467
LongHashmapBenchmark_findEntryFromCelixMap/10000/process_time/real_time              3.90 ns         3.89 ns    183397882 averageNrOfEntriesPerBucket=4.88281 items_per_second=256.477M/s nrOfBuckets=2.048k resizeCount=7 stdDeviationNrOfEntriesPerBucket=2.20062
LongHashmapBenchmark_findEntryFromCelixMap/100000/process_time/real_time             2.38 ns         2.38 ns    294279918 averageNrOfEntriesPerBucket=3.75601 items_per_second=420.113M/s nrOfBuckets=26.624k resizeCount=11 stdDeviationNrOfEntriesPerBucket=1.93968
LongHashmapBenchmark_findEntryFromCelixMap/1000000/process_time/real_time            2.62 ns         2.62 ns    270684694 averageNrOfEntriesPerBucket=4.98246 items_per_second=380.976M/s nrOfBuckets=200.704k resizeCount=28 stdDeviationNrOfEntriesPerBucket=2.23419
LongHashmapBenchmark_findEntryFromDeprecatedMap/10/process_time/real_time            3.67 ns         3.67 ns    190401851 items_per_second=272.19M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100/process_time/real_time           3.69 ns         3.68 ns    190481474 items_per_second=271.21M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000/process_time/real_time          3.69 ns         3.68 ns    189597715 items_per_second=271.084M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/10000/process_time/real_time         3.69 ns         3.68 ns    189618122 items_per_second=271.035M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100000/process_time/real_time        3.69 ns         3.68 ns    189681903 items_per_second=271.169M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000000/process_time/real_time       3.67 ns         3.67 ns    190523011 items_per_second=272.179M/s
```


#Hash 2, max load 5
```
-----------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                               Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------------------
LongHashmapBenchmark_findEntryFromStdMap/10/process_time/real_time                  0.428 ns        0.428 ns   1000000000 items_per_second=2.33786G/s
LongHashmapBenchmark_findEntryFromStdMap/100/process_time/real_time                 0.427 ns        0.427 ns   1000000000 items_per_second=2.34001G/s
LongHashmapBenchmark_findEntryFromStdMap/1000/process_time/real_time                0.428 ns        0.428 ns   1000000000 items_per_second=2.33579G/s
LongHashmapBenchmark_findEntryFromStdMap/10000/process_time/real_time               0.427 ns        0.427 ns   1000000000 items_per_second=2.3403G/s
LongHashmapBenchmark_findEntryFromStdMap/100000/process_time/real_time              0.427 ns        0.427 ns   1000000000 items_per_second=2.34037G/s
LongHashmapBenchmark_findEntryFromStdMap/1000000/process_time/real_time             0.428 ns        0.428 ns   1000000000 items_per_second=2.33835G/s
LongHashmapBenchmark_findEntryFromCelixMap/10/process_time/real_time                 2.56 ns         2.56 ns    272918800 averageNrOfEntriesPerBucket=0.625 items_per_second=390.34M/s nrOfBuckets=16 resizeCount=0 stdDeviationNrOfEntriesPerBucket=0.695971
LongHashmapBenchmark_findEntryFromCelixMap/100/process_time/real_time                3.76 ns         3.76 ns    186188393 averageNrOfEntriesPerBucket=3.125 items_per_second=266.203M/s nrOfBuckets=32 resizeCount=1 stdDeviationNrOfEntriesPerBucket=1.93245
LongHashmapBenchmark_findEntryFromCelixMap/1000/process_time/real_time               2.78 ns         2.78 ns    251731622 averageNrOfEntriesPerBucket=3.90625 items_per_second=359.86M/s nrOfBuckets=256 resizeCount=4 stdDeviationNrOfEntriesPerBucket=2.00366
LongHashmapBenchmark_findEntryFromCelixMap/10000/process_time/real_time              2.99 ns         2.99 ns    233585592 averageNrOfEntriesPerBucket=4.88281 items_per_second=333.894M/s nrOfBuckets=2.048k resizeCount=7 stdDeviationNrOfEntriesPerBucket=2.14352
LongHashmapBenchmark_findEntryFromCelixMap/100000/process_time/real_time             3.00 ns         3.00 ns    233887362 averageNrOfEntriesPerBucket=3.75601 items_per_second=333.838M/s nrOfBuckets=26.624k resizeCount=11 stdDeviationNrOfEntriesPerBucket=1.94674
LongHashmapBenchmark_findEntryFromCelixMap/1000000/process_time/real_time            2.99 ns         2.99 ns    234075293 averageNrOfEntriesPerBucket=4.98246 items_per_second=334.062M/s nrOfBuckets=200.704k resizeCount=28 stdDeviationNrOfEntriesPerBucket=2.22839
LongHashmapBenchmark_findEntryFromDeprecatedMap/10/process_time/real_time            3.64 ns         3.64 ns    192084107 items_per_second=274.697M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100/process_time/real_time           3.64 ns         3.64 ns    192187264 items_per_second=274.762M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000/process_time/real_time          3.64 ns         3.64 ns    192202353 items_per_second=274.977M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/10000/process_time/real_time         3.63 ns         3.63 ns    192491569 items_per_second=275.124M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100000/process_time/real_time        3.63 ns         3.63 ns    192638166 items_per_second=275.242M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000000/process_time/real_time       3.63 ns         3.63 ns    193028763 items_per_second=275.699M/s
```

# Hash3, max load 5
```
-----------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                               Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------------------
LongHashmapBenchmark_findEntryFromStdMap/10/process_time/real_time                  0.414 ns        0.414 ns   1000000000 items_per_second=2.4128G/s
LongHashmapBenchmark_findEntryFromStdMap/100/process_time/real_time                 0.414 ns        0.414 ns   1000000000 items_per_second=2.41719G/s
LongHashmapBenchmark_findEntryFromStdMap/1000/process_time/real_time                0.414 ns        0.414 ns   1000000000 items_per_second=2.41582G/s
LongHashmapBenchmark_findEntryFromStdMap/10000/process_time/real_time               0.414 ns        0.414 ns   1000000000 items_per_second=2.41681G/s
LongHashmapBenchmark_findEntryFromStdMap/100000/process_time/real_time              0.414 ns        0.414 ns   1000000000 items_per_second=2.41521G/s
LongHashmapBenchmark_findEntryFromStdMap/1000000/process_time/real_time             0.414 ns        0.414 ns   1000000000 items_per_second=2.41732G/s
LongHashmapBenchmark_findEntryFromCelixMap/10/process_time/real_time                 2.26 ns         2.26 ns    307909253 averageNrOfEntriesPerBucket=0.625 items_per_second=441.585M/s nrOfBuckets=16 resizeCount=0 stdDeviationNrOfEntriesPerBucket=0.856957
LongHashmapBenchmark_findEntryFromCelixMap/100/process_time/real_time                3.39 ns         3.39 ns    211113789 averageNrOfEntriesPerBucket=3.125 items_per_second=295.319M/s nrOfBuckets=32 resizeCount=1 stdDeviationNrOfEntriesPerBucket=1.86665
LongHashmapBenchmark_findEntryFromCelixMap/1000/process_time/real_time               2.48 ns         2.48 ns    281288298 averageNrOfEntriesPerBucket=3.90625 items_per_second=403.557M/s nrOfBuckets=256 resizeCount=4 stdDeviationNrOfEntriesPerBucket=2.16123
LongHashmapBenchmark_findEntryFromCelixMap/10000/process_time/real_time              3.31 ns         3.31 ns    211375732 averageNrOfEntriesPerBucket=4.88281 items_per_second=302.161M/s nrOfBuckets=2.048k resizeCount=7 stdDeviationNrOfEntriesPerBucket=2.2035
LongHashmapBenchmark_findEntryFromCelixMap/100000/process_time/real_time             2.48 ns         2.48 ns    282405576 averageNrOfEntriesPerBucket=3.75601 items_per_second=403.207M/s nrOfBuckets=26.624k resizeCount=11 stdDeviationNrOfEntriesPerBucket=1.92598
LongHashmapBenchmark_findEntryFromCelixMap/1000000/process_time/real_time            2.90 ns         2.90 ns    241719478 averageNrOfEntriesPerBucket=4.98246 items_per_second=344.381M/s nrOfBuckets=200.704k resizeCount=28 stdDeviationNrOfEntriesPerBucket=2.23011
LongHashmapBenchmark_findEntryFromDeprecatedMap/10/process_time/real_time            3.54 ns         3.54 ns    197602959 items_per_second=282.865M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100/process_time/real_time           3.54 ns         3.53 ns    197704646 items_per_second=282.885M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000/process_time/real_time          3.53 ns         3.53 ns    197684458 items_per_second=282.891M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/10000/process_time/real_time         3.54 ns         3.54 ns    197731536 items_per_second=282.88M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/100000/process_time/real_time        3.54 ns         3.54 ns    197896830 items_per_second=282.817M/s
LongHashmapBenchmark_findEntryFromDeprecatedMap/1000000/process_time/real_time       3.54 ns         3.54 ns    198110019 items_per_second=282.856M/s
```